### ISSUE TITLE
Don't swallow YAML parse errors

### ISSUE THREAD

**Issue by JanStevens** [AssetSync/asset_sync#331]
Hello,

If the YAML file cannot be parsed correctly (for whatever reason) AssetSync should just fail and die. We wasted a day on a type just to find out that AssetSync swallows any errors.

```
    def yml
      begin
        @yml ||= YAML.load(ERB.new(IO.read(yml_path)).result)[Rails.env] rescue nil || {}
      rescue Psych::SyntaxError
        @yml = {}
      end
    end
```

But after finding it it worked! (still happy camper)

Regards,

**Comment by PikachuEXE** [comment]
I find this commit only (with Blame)
https://github.com/AssetSync/asset_sync/commit/06af52e56482142d77a25fab5092e4dec11732ed
Not sure what's the reason of that change

Also already got #270 that proposed to NOT rescue the error
But I think I will create another PR to raise custom error with better error message

**Comment by PikachuEXE** [comment]
Fixed in e0084a2dcf9f3c2c34ba7ecbde4cda3e9e219066
